aws/eks-distro-build-tooling
# scaffolding for EKS Distro Build Tooling operations tool

*Issue #, if available:*
https://github.com/aws/eks-distro-internal/issues/250

*Description of changes:*
Scaffolding for an EKS Distro Build Tooling operational tool.

This adds:
- package structure
- makefile with example build targets for individual commands
- linter + unit test execution in makefile
- mocks support in Makefile
- logger cribbed from EKS-A

We can use this as the basis for further EKS-D Operational Tooling which requires us to go above-and-beyond shell scripting and make files.

Testing this out for yourself:
- check out this branch
- run `make build-eks-go-tools`
- run the eksGoTools binary that will be output to `./bin/darwin/$ARCH/eksGoTools` to see the command output

We can then flesh out the `pkg` with generalized operational tooling functionality (git & github interaction; ECR; S3; w/e) and build project and use-case specific cmds which utilize this generic functionality. 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
